operating_unit/models/res_users.py

```diff
@@ -14,6 +19,16 @@ def operating_unit_default_get(self, uid2=False):
         if not uid2:
             uid2 = self.env.user.id
         user = self.env["res.users"].browse(uid2)
```

Comment:
I would play it safer and just add a `sudo()` in this line:
```suggestion
        user = self.env["res.users"].sudo().browse(uid2)
```
Then you won't need to manually check if the user has permissions on either the model or specific records, and you can drop the rest of the `sudo()`s used below.